recipes/ohpipeline/all/patches/0003-ohpipeline.patch

```diff
@@ -0,0 +1,57 @@
+diff --git a/OpenHome/Av/Playlist/PinInvokerUpnpServer.cpp b/OpenHome/Av/Playlist/PinInvokerUpnpServer.cpp
+index a3919ec1..4e7afe42 100644
+--- a/OpenHome/Av/Playlist/PinInvokerUpnpServer.cpp
++++ b/OpenHome/Av/Playlist/PinInvokerUpnpServer.cpp
+@@ -20,6 +20,7 @@
+ 
+ #include <algorithm>
+ #include <atomic>
++#include <random>
+ #include <vector>
+ 
+ using namespace OpenHome;
+@@ -233,7 +234,9 @@ void PinInvokerUpnpServer::BrowseContainerCallback(IAsync& aAsync)
+     catch (XmlError&) {}
+ 
+     if (newContainers && iShuffle) {
+-        std::random_shuffle(iContainers.begin() + iContainersIndex, iContainers.end());
++        std::random_device rd;
++        std::mt19937 g(rd());
++        std::shuffle(iContainers.begin() + iContainersIndex, iContainers.end(), g);
+     }
+ 
+     if (iContainers.size() == iContainersIndex || playlistFull) {
+diff --git a/OpenHome/Av/Playlist/TrackDatabase.cpp b/OpenHome/Av/Playlist/TrackDatabase.cpp
+index 260cf6f0..caf883b0 100644
+--- a/OpenHome/Av/Playlist/TrackDatabase.cpp
++++ b/OpenHome/Av/Playlist/TrackDatabase.cpp
+@@ -8,6 +8,7 @@
+ #include <OpenHome/Av/Debug.h>
+ 
+ #include <algorithm>
++#include <random>
+ #include <vector>
+ 
+ using namespace OpenHome;
+@@ -391,7 +392,9 @@ Track* Shuffler::NextTrackRef(TUint aId)
+                 else if (index == iShuffleList.size()-1) {
+                     // we've run through the entire list
+                     // prefer re-shuffling over repeating the order of tracks if we play again
+-                    std::random_shuffle(iShuffleList.begin(), iShuffleList.end());
++                    std::random_device rd;
++                    std::mt19937 g(rd());
++                    std::shuffle(iShuffleList.begin(), iShuffleList.end(), g);
```

Comment:
The old warrior `std::random_shuffle` was deprecated in C++14 and removed in C++17, but the upstream did change it, right? Could please open a PR to the upstream with this change? Usually we do not accept this kind of portability, because the upstream is not C++17 tested as well. I still prefer keeping C++17 not supported with `check_max_cppstd` method, e.g https://github.com/conan-io/conan-center-index/blob/master/recipes/exiv2/all/conanfile.py#L108

Reply:
The upstream uses C++11. Clang on Mac uses 14 by default, I think and that's why the change was required.

Changes CMake to use C++11 so this should no longer be necessary.